src/collections/lazyCollection.ts

```diff
@@ -21,6 +21,7 @@ export class LazyCollection<T> implements Iterable<T> {
     const iter = this.getIterator();
     let next = iter.next();
     while (!next.done) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
```

Comment:
nit: could we have done anything to avoid these eslint exceptions?  Is the issue that the value could be undefined, but is defined as only taking T?

Reply:
Maybe, but nothing I could figure out.

I tried a few things, but `this.getIterator` has an any in its type 
`(method) LazyCollection<T>.getIterator(): Iterator<T, any, undefined`